test_scripts/Defects/5_0/1904_Media_app_activation_during_active_embedded_audio_source.lua

```diff
@@ -0,0 +1,79 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/1904
+--
+-- Description:
+-- Media app activation during active embedded audio source
+-- Precondition:
+-- SDL and HMI are started.
+-- Media app is registered.
+-- Media app in BACKGROUND and NOT_AUDIBLE due to active embedded audio source.
+-- In case:
+-- 1) User activates this media app and SDL receives from HMI:
+-- a) OnEventChanged (AUDIO_SOURCE, isActive=false)
+-- b) SDL.ActivateApp (<appID_of_communication_app>)
+-- Expected result:
+-- 1) SDL must respond SDL.ActivateApp (SUCCESS) to HMI send OnHMIStatus (FULL, AUDIBLE) to mobile app.
+--    Information:
+--    Expected HMI behavior:
+-- a. Media app activation is the trigger for HMI to switch off embedded audio source
+-- b. HMI switches off embedded audio source and sends OnEventChanged (AUDIO_SOURCE, isActive=false) to SDL
+-- Actual result:
+-- SDL does not set required HMILevel and audioStreamingState.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+config.application1.registerAppInterfaceParams.appHMIType = { "MEDIA" }
+config.application1.registerAppInterfaceParams.isMediaApplication = true
+
+--[[ Local Functions ]]
+local function deactivateApp()
+    common.getHMIConnection():SendNotification("BasicCommunication.OnAppDeactivated", { appID = common.getHMIAppId() })
+    common.getMobileSession():ExpectNotification("OnHMIStatus", {
+        hmiLevel = "LIMITED", audioStreamingState = "AUDIBLE"
+    })
+end
+
+local function embeddedAudioSource()
+    common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged",
+    {eventName = "AUDIO_SOURCE", isActive = true})
+
+    common.getMobileSession():ExpectNotification("OnHMIStatus",
+    {hmiLevel = "BACKGROUND", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+end
+
+local function activateMediaApp()
+    common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged",
+    {eventName = "AUDIO_SOURCE", isActive = false})
+    
+    common.getMobileSession():ExpectNotification("OnHMIStatus",
+        {hmiLevel = "LIMITED", audioStreamingState = "AUDIBLE"},
+        {hmiLevel = "FULL", audioStreamingState = "AUDIBLE" })
+        :Times(2)
+        :Do(function(exp)
+            if exp.occurences == 1 then
+            local requestId = common.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = common.getHMIAppId() })
```

Comment:
`SDL.ActivateApp` request must be sent right after `BasicCommunication.OnEventChanged`, response also wait after the sending request